addons/project/models/project.py

```diff
@@ -858,6 +858,17 @@ def _read_group_stage_ids(self, stages, domain, order):
     repeat_show_week = fields.Boolean(compute='_compute_repeat_visibility')
     repeat_show_month = fields.Boolean(compute='_compute_repeat_visibility')
 
+    # Account analytic
+    analytic_account_id = fields.Many2one('account.analytic.account', ondelete='set null',
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]", check_company=True,
+        help="Analytic account to which this task is linked for financial management. "
+             "Use an analytic account to record cost and revenue on your task. "
+             "If empty, the analytic account of the project will be used.")
+    task_analytic_account_id = fields.Many2one('account.analytic.account', compute='_compute_task_analytic_account_id')
+    project_analytic_account_id = fields.Many2one('account.analytic.account', string='Project Analytic Account', related='project_id.analytic_account_id')
+    analytic_tag_ids = fields.Many2many('account.analytic.tag',
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]", check_company=True)
```

Comment:
FYI, fields specified as `check_company=True` have a domain set by default (if no domain is specified).
I don't say you should remove your specified domain, since the default domain is quite magic and sometimes it makes sense to prefer explicit domains than implicit ones, it's just to give some details on the check_company feature ^^.

https://github.com/odoo/odoo/blob/master/odoo/fields.py#L2609-L2625